File: app/src/main/java/com/hjc/base/utils/web/MyWebViewClient.java
package com.hjc.base.utils.web;

import android.webkit.WebView;
import android.webkit.WebViewClient;

/**
 * @Author: HJC
 * @Date: 2019/3/8 11:15
 * @Description: 自定义WebViewClient
 */
public class MyWebViewClient extends WebViewClient {

    @Override
    public boolean shouldOverrideUrlLoading(WebView view, String url) {
        view.loadUrl(url);
        return super.shouldOverrideUrlLoading(view, url);
    }

    // 视频全屏播放按返回页面被放大的问题
    @Override
    public void onScaleChanged(WebView view, float oldScale, float newScale) {
        super.onScaleChanged(view, oldScale, newScale);
        if (newScale - oldScale > 7) {
            //异常放大，缩回去
            view.setInitialScale((int) (oldScale / newScale * 100));
        }
    }
}

File: WebViewLib/src/main/java/com/hjc/webviewlib/client/MyWebViewClient.kt
package com.hjc.webviewlib.client

import android.graphics.Bitmap
import android.widget.ProgressBar
import com.tencent.smtt.sdk.WebView
import com.tencent.smtt.sdk.WebViewClient

/**
 * @Author: HJC
 * @Date: 2021/1/9 15:31
 * @Description: 自定义WebViewClient
 */
class MyWebViewClient(private val mProgressBar: ProgressBar) : WebViewClient() {

    // 防止加载网页时调起系统浏览器
    override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
        view.loadUrl(url)
        return true
    }

    override fun onPageStarted(webView: WebView, s: String, bitmap: Bitmap) {
        super.onPageStarted(webView, s, bitmap)
        mProgressBar.visibility = WebView.VISIBLE
    }

    override fun onPageFinished(webView: WebView, s: String) {
        super.onPageFinished(webView, s)
        mProgressBar.visibility = WebView.GONE
    }
}